drivers/timer/apic_timer.c

```diff
@@ -211,6 +211,11 @@ uint32_t sys_clock_cycle_get_32(void)
 	return ret;
 }
 
+uint64_t sys_clock_cycle_get_64(void)
+{
+	return 0;
```

Comment:
@nashif - would be great to get your input here. It seems there is no 64-bit timer register on this platform.

For software emulation, I would like to increment a 32-bit "top half" each time a 32-bit cycle counter overflows (assuming a 1:1 ratio), shift and then OR it with the regular 32-bit value.

In this driver, where is the best location to increment the 32-bit "top half"?

Reply:
Actually on all modern hardware the APIC LVT timer is slaved to the TSC (with an configurable shift), so it's possible to synthesize a 64 bit value.  That's probably not worth it though.

I'm definitely not a fan of adding runtime complexity to a timer driver just to emulate a rarely-needed feature.  The overwhelmingly common use of cycle timing is to compute tiny delays, after all.  It's not worth taking a performance or code size penalty on hardware just to support oddball use cases.

I guess my preference here would be to make the 64 bit cycle API an optional thing, gated by a kconfig.  You can use the API if your timer hardware supports it, and not if it doesn't.  That's more or less the way it works for ticks (c.f. CONFIG_TIMEOUT_64BIT), though the concern there is code size and not hardware capability as the kernel is willing to do the wrap logic internally.